### PR TITLE
add news

### PR DESCRIPTION
**Pull request by ruslandoga**
https://github.com/getsince/test5/blob/45ddea19ededa4f3823a68ee301ce202ff044dbb/Since/App/AppSocket.swift#L35 would need to pass a `"locale"` for gettext

### CODE REVIEW

**Comment by akhrail1996** [comment]
> https://github.com/getsince/test5/blob/45ddea19ededa4f3823a68ee301ce202ff044dbb/Since/App/AppSocket.swift#L35 would need to pass a `"locale"` for gettext

https://github.com/getsince/test5/pull/209/commits/b64707e34ac8ac3b5f7e6e80a9a106ca0945e588

File: lib/t_web/channels/feed_channel.ex [hunk 1]
```diff
@@ -128,6 +135,11 @@ defmodule TWeb.FeedChannel do
     {:reply, :ok, socket}
   end
 
+  def handle_in("seen-news", _params, socket) do
```
[the remark this answers is not in the document]
Comment (ruslandoga) [reply lib/t_web/channels/feed_channel.ex:138]: I assume that'd be only one news story at a time. I'll add support for multiple stories tomorrow.

```swift
struct NewsStory {
  let id: Int
  let story: [StoryPage]
}

channel.push("seen", ["news": story.id]) { ... }
```

[the remark this answers is not in the document]
Comment (akhrail1996) [reply lib/t_web/channels/feed_channel.ex:138]: actually it currently works with multiple stories already:
image
(this works)

File: lib/t_web/channels/feed_channel.ex [hunk 2]
```diff
@@ -128,6 +135,11 @@ defmodule TWeb.FeedChannel do
     {:reply, :ok, socket}
   end
 
+  def handle_in("seen", %{"news_story_id" => news_story_id}, socket) do
```
Comment (ruslandoga) [inline lib/t_web/channels/feed_channel.ex:138]: this has changed

File: test/t_web/channels/feed_channel_test.exs [hunk 3]
```diff
@@ -433,6 +433,48 @@ defmodule TWeb.FeedChannelTest do
...
+      {1, _} =
+        News.SeenNews
+        |> where(user_id: ^me.id)
+        |> Repo.delete_all()
+
+      assert {:ok, %{"news" => news}, socket} = join(socket, "feed:" <> me.id)
+
+      assert news == [
+               _first_news_item = %{
+                 id: 1,
```
Comment (ruslandoga) [inline test/t_web/channels/feed_channel_test.exs:449]: now both `id` and `story` are returned, not just `story`

File: test/t_web/channels/feed_channel_test.exs [hunk 4]
```diff
@@ -433,6 +433,48 @@ defmodule TWeb.FeedChannelTest do
...
+                         "center" => [142.33, 317.83]
+                       }
+                     ],
+                     "size" => [428, 926]
+                   }
+                 ]
+               }
+             ]
+
+      ref = push(socket, "seen", %{"news_story_id" => 1})
```
Comment (ruslandoga) [inline test/t_web/channels/feed_channel_test.exs:466]: once the user scrolled past the news story, it should be `"seen"`

**Review by akhrail1996 — APPROVED** [review]
👍

**Pull request merged**